### DISCUSSION TITLE
Troubleshooting

### DISCUSSION THREAD

**Discussion by TwoPigsToTheWind** [slu4coder/Minimal-64x4-Home-Computer#22] [Q&A]
recently finished assembly. Have already checked for broken pins, backwards, or misplaced ICs and rewrote the 4 flash ICs. Any suggestions as to what to check next?

![16a55769-2d2b-4e7d-b527-ada26c09583b](https://github.com/slu4coder/Minimal-64x4-Home-Computer/assets/169104967/e1cda3dc-2554-4eb5-9dbb-144e209554c0)

**Comment by nateo87** [comment]
(Apologies if this is was already apparent)  did you press reset?

**Reply by TwoPigsToTheWind** [reply]
Happy to say I did try the reset button. When the reset button is pressed the screen gets slightly brighter and on some resets a diagonal run of pixels will shift. In roughly a fifth of power on events a very regular pattern of straight lines will display.

The same snow type screen as pictured above is seen if the system is powered on with the ssd flash removed which leads me to think the problem is related to address or data paths running to and from the ssd.

**Comment by joelburton** [comment]
The static-image you're seeing pre-reset appears on a working system, so
that isn't an indication of a problem. The snow is just an accurate
representation of "there's junk in the video memory before reset".

That you're seeing that *probably* means the VGA-related stuff is good (you
have valid sync, etc), so the problem is probably on the computer-logic-end
of the board.

If you connect via UART, do you get anything?

On Sun, Jun 30, 2024 at 9:17 PM TwoPigsToTheWind ***@***.***>
wrote:

> Happy to say I did try the reset button. When the reset button is pressed
> the screen gets slightly brighter and on some resets a diagonal run of
> pixels will shift. In roughly a fifth of power on events a very regular
> pattern of straight lines will display.
>
> The same snow type screen as pictured above is seen if the system is
> powered on with the ssd flash removed which leads me to think the problem
> is related to address or data paths running to and from the ssd.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/slu4coder/Minimal-64x4-Home-Computer/discussions/22#discussioncomment-9920999>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAI7XFWAN662O762AL6JV2LZKDJ6JAVCNFSM6AAAAABKA2L7ZCVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TSMRQHE4TS>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID:
> <slu4coder/Minimal-64x4-Home-Computer/repo-discussions/22/comments/9920999
> @github.com>
>


-- 
Joel Burton - ***@***.***

GTalk: ***@***.***, Mobile/SMS: +1 415 513 2470

**Comment by joelburton** [comment]
some core things to double-check:
- are all the chips in the correct alignment; there are a number of flipped
chips
- there are 74HC02 chips and the 74HC4002 chip on the board, and those are
different: are you sure you have those differentiated?
- what's the voltage between GND and 5V? (should be close to 5.0)
- how much power is the board drawing? mine is ~1.0 W

**Reply by TwoPigsToTheWind** [reply]
voltage looks good, drawing .15amps so right at 1w power, that does decrease to .11 when the reset is pressed before recovering. chip placement has been checked by two sets of eyes. thinking at this point it is either a bad chip or a bad solder joint.

**Comment by joelburton** [comment]
Can you connect via UART? (500_000 baud, 2 stop bits, it should say
"READY." after reset coming out)

On Mon, Jul 1, 2024 at 3:40 PM TwoPigsToTheWind ***@***.***>
wrote:

> voltage looks good, drawing .15amps so right at 1w power, that does
> decrease to .11 when the reset is pressed before recovering. chip placement
> has been checked by two sets of eyes. thinking at this point it is either a
> bad chip or a bad solder joint.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/slu4coder/Minimal-64x4-Home-Computer/discussions/22#discussioncomment-9930284>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAI7XFQALOZSVQRKNNAJJKDZKHLEZAVCNFSM6AAAAABKA2L7ZCVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TSMZQGI4DI>
> .
> You are receiving this because you commented.Message ID:
> <slu4coder/Minimal-64x4-Home-Computer/repo-discussions/22/comments/9930284
> @github.com>
>


-- 
Joel Burton - ***@***.***

GTalk: ***@***.***, Mobile/SMS: +1 415 513 2470

**Reply by TwoPigsToTheWind** [reply]
whole lot of nothing on the UART

**Comment by joelburton** [comment]
I'd draw the inference that if you get the expected-snow-pre-reset, get strangeness (sometimes snow, sometimes-lines) post-reset, and don't get anything on UART, it's most likely that the "CPU" area is where you have a problem (so: prob not video chips or UART chips).

Some things I'd suggest checking:
- what's the output on CLK (there's a pinout on the extension header); if you have a logic analyzer, you could see the square wave. If not, your multimeter should show ~2.5V (since it cycles too fast for your multimeter, it should show an average of 0/5v)
- take a look at the RESET signal on the header so you can see if that's cleanly low-during-button-push and high-afterwards. If this bounces a bit, it will prob be hard to see that on a multimeter, I fear
- with the board powered on, check the 5v and GND pins on your chips to see if they're all getting ~5v (sometimes a dead chip will show some other voltage difference)

I've got a working machine; if there's anything I can measure for you for you to check on your machine, I'm happy to.

**Reply by TwoPigsToTheWind** [reply]
all great ideas

- CLK on the extension header reads about 2.8V 
- RESET on the header displays low while pressed high when open, if there is some bounce I agree my meter won't catch it
- voltage across each chip reads between 4.7-4.9 volts 

I think I'll just have to check thru each net for continuity, must be a bad joint in there somewhere

**Reply by TwoPigsToTheWind** [reply]
@joelburton what resistance to you measure across the 5v and ground header pins on the 2x3 power connector when the power is not connected?

**Reply by joelburton** [reply]
~2 megs, @TwoPigsToTheWind

**Reply by TwoPigsToTheWind** [reply]
Thanks @joelburton, I'm getting about 30, maybe there's a bad cap

**Reply by TwoPigsToTheWind** [reply]
@joelburton pin 22 on U31 and pin 30 on U43 are reading 5V when power if first connected and then 2.5V at after reset. I assume the 2.5V is what I"m seeing as the average of the high and low as the U34 RAM and U36 VRAM take turns. Does that look about the same on yours? Listed bellow are some other voltages coming off the expansion header, maybe if we can identify a discrepancy it will help narrow down the issue. The power on figures may be the more useful ones for comparison. Once your machine is reset and starts running code properly the average voltage(frequency of state change) of the pins listed bellow is likely dependent on the code being run and should differ from my stalled out non-booting system.  

RO 1.5 @ power on & after Reset
RI 4.2  @ power on & after Reset
A15 3.9 @ power, 2.5 after Reset
A14 1 @ power, 2.5 after Reset
A13 1 @ power, 2.5 after Reset
A12 1 @ power, 2.5 after Reset
A11 1 @ power, 2.5 after Reset
A10 1 @ power, 2.5 after Reset
A9 1 @ power, 2.5 after Reset
A8 1 @ power, 2.5 after Reset
A7 3.6 @ power, 2.0 after Reset
A6 1.2 @ power, 2.0 after Reset
A5 1.2 @ power, 2.0 after Reset
A4 1.2 @ power, 3.8 after Reset
A3 1.2 @ power, 3.5 after Reset
A2 1.2 @ power, 3.3 after Reset
A1 1.2 @ power, 2.5 after Reset
A0 2.3 @ power on & after Reset
BUS7 4.8 @ power on, 3.3 after Reset
BUS6 0 @ power on, 3.0 after Reset
BUS5 0 @ power on, 3.1 after Reset
BUS4 0 @ power on, 3.3 after Reset
BUS3 0 @ power on, 3.3 after Reset
BUS2 0 @ power on, 3.0 after Reset
BUS1 0 @ power on, 2.75 after Reset
BUS0 0 @ power on, 1.85 after Reset
RES 4.8 @ power on, 0 during Reset, 4.8 after 
INH 0 @ power on & after Reset

**Comment by TwoPigsToTheWind** [comment]
@joelburton I picked up a 4ch scope this weekend, any suggestion on what to probe first? Picking this project back up after a few months away.

**Comment by slu4coder** [comment]
Hi Joel, although the community has built quite a number of Minimal 64x4s successfully now, there have been issues with "budget" ICs. The design uses the ICs to their full capability, so timing becomes critical quite easily. I think it makes sense to start further debugging after you made sure you have only used new parts from quality suppliers (like Mouser or such) for your built. From the failure mode you have posted, I'd guess the CPU gets stuck during the boot process copying its OS code from FLASH to RAM. I'd start with swapping U24, U19, U6, U7, U2. Good luck!

**Reply by TwoPigsToTheWind** [reply]
@slu4coder  Certainly up for swapping the ICs listed above, looks like two of the '161s, a '245, and two '32. All the parts on the board are from Mouser or Digikey so generally quality suppliers. There's always a chance that one or more of the ICs have some ESD damage. Might just need to get a chip tester to look for duds. 

Are there and probeable signals related to the boot process you'd suggest I take a look at under the scope to try and narrow down the source of the issue?

**Reply by slu4coder** [reply]
Unfortunately, I can't point you to a specific signal. It is more the relative timing between different signals that may cause problems. Swapping some ICs seems a good starting point.

**Reply by hellboy73** [reply]
Hi everyone
I have similar issue. Even after and during reset the pattern is not changing.
![480501782_987280426381858_592363159484053259_n](https://github.com/user-attachments/assets/0dd9a02d-b4f6-480f-ac7e-ed9cc0ce451e)

Of course I initially ordered cheap random parts, but now I learned my lesson and I'm going to replace all of them. 
My question is which type should I choose:
CD74HCxxxE or SN74HCxxxN ?
as Im very new to electronics I don't see the difference. 
Thanks

**Reply by hellboy73** [reply]
Just figured, CD74 and SN74 are in fact different factory codes even though the chips come from TI. So i guess any of those should be fine.

**Reply by hellboy73** [reply]
Today i replaced all the chips with new ones from Texas Instruments ordered from trusted supplier. The effect is the same, vertical bars with some snow. Reset doesn't change anything beside the fact, power consumption drops from 0.2A to 0.15A
Again carefully inspected al the solder joints, reheated most of them just to be sure. I have double checked all the resistors and again  checked IC orientation. 
No idea what to check next, apart from the fact I will probably need UART interface and some kind of logic analyzer.
![481484476_2476909562650243_1709617508511459069_n](https://github.com/user-attachments/assets/9a192e65-7739-42be-ae81-bcfedb9a03d5)

**Comment by TwoPigsToTheWind** [comment]
Does anyone have an image of what the 8mhz clock singal looks like on a scope from a working minimal64x4 board? I'll post a pic of how mine looks tomorrow but it's a bit more irregular in shape than I would have expected when I was probing around this past weekend.

**Comment by slu4coder** [comment]
Hi, the random garbage is okay during pre-startup. The vertical stripes worry be a bit. There is one spot to look for:

It is critical for the CPU to have the *inputs* of the control ROMs change all at the same time. If not, this results in spikes on the control ROM's outout. Sensitive are EC, ES for example.

These control ROM inputs are defined by the outputs of the 74HC161 step counter (U24), the 74HC377s IR (U10) and FR (U5). Since these are different ICs they are potentially *not in sync* and need adjustment. This is done by R27 -- a RC timing constant. In the schematic, R27 is given as "zero ohms". Change this to 200 or even 400Ohms and observe EC and ES.

**Reply by TwoPigsToTheWind** [reply]
I'm going to try switching out R27 on my board which currently only outputs random garbage pre and post reset. Might as well. Pretty sure my issue is either a bad chip or a bad joint but I might as well try switching out R27 before building out a whole new board.

**Reply by hellboy73** [reply]
I'm getting exactly same stripes with R27=0, 100, 220, 330, 550, 650 Ohm
even with R27 disconnected the result is the same.

**Comment by hellboy73** [comment]
This is the closest i was to running Minimal64x4 so far. After soldering 2 independent boards to rule out any bad joint, with all the brand new chips.
![IMG_7778](https://github.com/user-attachments/assets/c629e5c9-bddb-43fc-a7fd-1fd803f8c1d1)
I had this screen for about 2 minutes. power consumption was at 0.17A 
Then suddenly screen got black and power consumption increased to 0.45A. 
Now no matter what i do, even after disconnecting power and booting again, all i get is black screen. 
Any suggestion what to check next? looks like VGA part had an issue and then broke, right?

**Reply by hellboy73** [reply]
I just discovered U40 74HC574 went bad and started heating, replaced it (second time) and I'm back to this boot screen with regular vertical stripes adding extra pixel every 8th pixel...
With keyboard connected everything works, even invaders. However still I have those vertical stripes making the whole screen wider by this extra pixel.

**Reply by hellboy73** [reply]
Finally I can say, I’m a proud owner of working Minimal64x4. It was a bumpy road to success :) 
Used two boards just to rule out any issues with my soldering. I had multiple issues with different RAM chips, only brand new ones work for me. Others give different types of stripes or garbage on screen. Two times U40 failed on me, not sure what was the reason but the third one (from a cheap reseller) works without issues till now. 
My monitor is not dealing properly with Minimal screen mode, it detects it correctly but the image is too wide. Old LCD TV with VGA input gives the correct image. And last but not least bad solder joint on VRAM caused the vertical stripes. I’m very new to electronics, but fortunately most of the issues with chips I was able to detect by hand finding the warmer ones. 
Now the fun begins.

![481113197_676346034814027_2148784695978910248_n](https://github.com/user-attachments/assets/e42a7390-abc5-4b81-89bc-217c6d40ee2a)

**Reply by slu4coder** [reply]
Hi, I realize, I am totally late to the game but congrats for getting your Minimal to work anyways! Looks great. After some "detour" into very different aspects of daily life I am slowly getting back into the Minimal now ;-)
